xapian-applications/omega/docs/newformat.rst

```diff
@@ -0,0 +1,174 @@
+
+  file --mime-type example.fb2
+
+which responses::
+
+  example.fb2: text/xml
+
+Sometimes ``file`` just returns a generic answer (most commonly ``text/plain`` or ``application/octet-stream``) and occasionally it misidentifies a file. If that is the case, we can associate the file format extension with a particular mime type at 'mimemap.tokens'. If multiple extensions are used for a format (such as htm and html for HTML) then add an entry for each.
+
+When indexing a filename which has an extension in upper-case or mixed-case, omindex will check for an exact match for the extension, and if not found, it will force the extension to lower-case and try again, so just add the extension in lower-case unless different cases actually have different meanings.
+
+In this example, ``text/xml`` is too broadly so we can associate ``fb2`` to ``application/x-fictionbook+xml`` which is much more specific.
+
+Then, you have to decide if you will use a library or an external filter.
+
+Using a library:
+================
+
+For safety reasons, it is not allowed to directly add libraries to omega source code. It is possible that some libraries contain bugs that affect the correct operation of the program. Because of that, omega isolates external libraries in subprocesses using modules ``worker`` and ``assistant``. These modules will take care of the safety measures and use handlers to get access to the libraries.
+
+1. To begin with, we need a library to extract information from the desired format. If many options are available, try to choose a library with an active community, that supports UTF-8 encoding and has a licence compatible with MIT/X license and GNU GPL version 2 and later. If UTF-8 output is not supported, you can convert the encoding using ``convert_to_utf8`` (replacing "ISO-8859-1" with the character set which is produced)
+   ::
+
+     convert_to_utf8(text, "ISO-8859-1");
+
+2. Once you have the library and the mimetype, it is time to modify the code. We have to create a new handler, which is the process used by omindex to access to the library. In order to do it, we have to create a file ‘handler_yourlibrary.cc’ that includes ‘handler.h’ and gives a definition to function ``extract`` (there are some examples at xapian-applications/omega such as 'handler_poppler.cc'). For this, inside the function body you will use the library to get the necessary information and store it in the corresponding arguments
+   ::
+
+     bool
+     extract(const std::string& filename,
+             std::string& dump,
+             std::string& title,
+             std::string& keywords,
+             std::string& author,
+             std::string& pages,
+             std::string& error)
+
+  Error messages should be written in ``error`` (you wont be able to write in the standard input/output) and you should fill the rest of the arguments with the information extracted from the document (It is not necessary to fill in all the arguments).
```

Comment:
"won't" ("wont" is also an English word, but not the one you want here)

"write to standard input/output" reads better.  But actually you mean "output/error" (you wouldn't expect to be able to write to standard input anyway).

Reply:
Okay, I am going to change it.